cra-scripts/bito-cra.sh

```diff
@@ -580,9 +584,15 @@ for param in "${required_params[@]}" "${bee_params[@]}" "${optional_params[@]}";
     elif [ "$param" == "secret_scanner_feedback" ]; then
         props[$param]=$(validate_boolean "${props[$param]}")
         docker_cmd+=" --secret_scanner_feedback=${props[$param]}"
+    elif [ "$param" == "acceptable_suggestions_enabled" ]; then
+        props[$param]=$(validate_boolean "${props[$param]}")
+        docker_cmd+=" --acceptable_suggestions_enabled=${props[$param]}"
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider consolidating boolean parameter handling</b></div>
<div id="fix">

Consider consolidating the boolean parameter handling into a helper function to reduce code duplication. The pattern `validate_boolean` followed by setting docker command is repeated multiple times. A similar issue was also found in cra-scripts/bito-cra.ps1 (line 570-581).
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -584,15 +584,7 @@
  function handle_boolean_param() {
    local param=$1
    local value=$2
    props[$param]=$(validate_boolean "$value")
    docker_cmd+=" --$param=${props[$param]}"
  }

      elif [ "$param" == "secret_scanner_feedback" ]; then
 -        props[$param]=$(validate_boolean "${props[$param]}")
 -        docker_cmd+=" --secret_scanner_feedback=${props[$param]}"
 -    elif [ "$param" == "acceptable_suggestions_enabled" ]; then
 -        props[$param]=$(validate_boolean "${props[$param]}")
 -        docker_cmd+=" --acceptable_suggestions_enabled=${props[$param]}"
 +        handle_boolean_param "$param" "${props[$param]}"
 +    elif [ "$param" == "acceptable_suggestions_enabled" ]; then
 +        handle_boolean_param "$param" "${props[$param]}"
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/gitbito/CodeReviewAgent/pull/46#issuecomment-2536074802>#9f0e9d</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged